### PR TITLE
Add text-ellipsis utility class.

### PR DESCRIPTION
**Pull request by cannandev**
## Changes proposed in this pull request:

Create a reusable utility class that adds an ellipsis at the end of a block of text. The new dashboard design uses ellipses often. We want to use CSS to change the visual display, without affecting screen reader text.

image

### How to use
Apply `text-ellipsis` class to any element. It will add an three dots as the text approaches the end of its container. 

⚠️ Note: The attribute `text-overflow` relies on the element's display set to `block`. For text that has another element following it on the same line, set the parent display to `flex`. 

Example: 
```tsx
<header className="display-flex">
  <strong className="orgs-selector__current text-bold text-base-dark text-ellipsis">
    sandbox-gsa-much-longer-name-goes-here-and-is-very-very-long
  </strong>
  <button>toggle</button>
</header>
```

### Related issues
#409 

### Submitter checklist

- [ ] Added logging is not capturing sensitive data and is set to an appropriate level (DEBUG vs INFO etc)
- [ ] Updated relevant documentation (README, ADRs, explainers, diagrams)

### Security considerations

None. CSS only. 

### Acceptance criteria

- [ ] Ellipsis appears on text longer than its container.
- [ ] Full text is still announced by screen readers.

### CODE REVIEW

File: src/assets/stylesheets/styles.scss [hunk 1]
```diff
@@ -100,17 +100,30 @@
...
 }
 .text-break-all {
   word-break: break-all;
 }
+.text-ellipsis {
+  @include u-overflow('hidden');
+  @include u-text('no-wrap');
+  @include u-display(
+    'block'
+  ); // Use flex display on parent for inline elements.
```
Comment (beepdotgov) [inline src/assets/stylesheets/styles.scss:116]: Non-blocking question: is this whitespace significant to the mixin? (Just asking for my edification here, I’m only curious!)

File: src/assets/stylesheets/styles.scss [hunk 2]
```diff
@@ -100,17 +100,30 @@
...
+  text-overflow: clip;
```
Comment (beepdotgov) [inline src/assets/stylesheets/styles.scss:117]: Non-blocking question: is there a reason to set both `clip` and `ellipsis`? (Like, is `clip` included here as a fallback?)

**Review by beepdotgov — DISMISSED** [review]
Looks great to me! Left two non-blocking questions, but that’s it — so excited to see this in action, @cannandev 🎉

File: src/assets/stylesheets/styles.scss (hunk 2 above)
Comment (cannandev) [reply src/assets/stylesheets/styles.scss:117]: `clip` is including as the fallback.

Comment (cannandev) [reply src/assets/stylesheets/styles.scss:117]: ....However, [looking at the spec](https://developer.mozilla.org/en-US/docs/Web/CSS/text-overflow#browser_compatibility), `ellipsis` is supported by the same number of devices and browsers and `clip`. I'll go ahead and remove `clip`.  Thanks for asking! Save a line of code.

Comment (beepdotgov) [reply src/assets/stylesheets/styles.scss:117]: Nice! And same, thanks for talking through it with me!

File: src/assets/stylesheets/styles.scss (hunk 1 above)
Comment (cannandev) [reply src/assets/stylesheets/styles.scss:116]: Great question! The text-overflow property doesn't force an overflow to occur by itself. To make text overflow its container, you have to set other CSS properties `overflow` and `white-space`. 

For _my_ edification, I tried setting the white-space to default `wrap` and this happened 😬 
image

Comment (beepdotgov) [reply src/assets/stylesheets/styles.scss:116]: Thanks @cannandev! And sorry, I don’t think my question was remotely clear: I was wondering if there’s something about the `u-display` mixin that has to be written like this, across multiple lines:

```@include u-display(
    'block'
  );```

(The previous two mixins are all written on one line, so I was just curious.) (And again: non-blocking question!)

File: src/assets/stylesheets/styles.scss (hunk 2 above)
Comment (cannandev) [reply src/assets/stylesheets/styles.scss:117]: @beepdotgov Also, if you wrote this spec, ignore me.

Homer disappearing into the bushes

Comment (beepdotgov) [reply src/assets/stylesheets/styles.scss:117]: @cannandev oh hey they don’t let me anywhere near specs, have you _seen_ me write css

Homer Simpson sits in the middle of a room filled with impressive array of lights, buttons, and levers, spinning his chair around haphazardly.

**Comment by jduss4** [comment]
[non-blocking question / suggestion] What are your thoughts on adding an example of this to the design-guide page? Maybe that's coming once it's tied to the dropdown, but I could see it being valuable for someone figuring out the right combination of CSS classes that need to exist for the ellipses to work!

**Review by echappen — DISMISSED** [review]
I just tested this utility in my work on the users list, and it works swimmingly. LGTM!

**Comment by echappen** [comment]
@cannandev Please ff to remove the `.text-ellipsis` selector that I was using provisionally.

File: src/assets/stylesheets/styles.scss (hunk 1 above)
Comment (cannandev) [reply src/assets/stylesheets/styles.scss:124]: It's the prettier linter that runs on save. It sees the comment as part of the line length. 😞

**Comment by cannandev** [comment]
> @cannandev Please ff to remove the `.text-ellipsis` selector that I was using provisionally.

@echappen Can you approve the PR again so I can merge it! Thank you.

**Review by echappen — APPROVED** [review]

**Pull request merged**